browser/importer/chrome_profile_lock.cc

```diff
@@ -22,12 +23,14 @@ ChromeProfileLock::~ChromeProfileLock() {
 }
 
 void ChromeProfileLock::Lock() {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
   if (HasAcquired())
     return;
   lock_acquired_ = process_singleton_->Create();
```

Comment:
shouldn't we be using `ChromeProcessSingleton` here so we also handle the startup lock?

Reply:
I did some code review and I think ProcessSingleton alone is all that's necessary to check for/obtain exclusive access to the Chrome user data directory. The ProcessSingletonStartupLock seems designed to handle queuing and replaying command line invocations that occur while the startup lock is locked, but we don't actually want to do that in Brave when importing from a Chrome user data directory—we just want any such invocations of a Chrome process with the user data directory being imported to fail until the import is complete; it doesn't make sense for Brave to then try to handle those invocations, since they are meant for Chrome and not Brave.

Reply:
I was wondering if there might be a race condition if the import ran during Chrome startup. Can anything modify the user data dir during the startup lock?

Reply:
> Can anything modify the user data dir during the startup lock?

I think ProcessSingleton's [lock file](https://cs.chromium.org/chromium/src/chrome/browser/process_singleton_posix.cc?l=1011-1012&rcl=dc98dbfd177787df0d527e2f131c00f82b03e238) is intended to prevent concurrent modification of the user data directory.